roles/ceph-osd/tasks/main.yml

```diff
@@ -8,21 +8,21 @@
 
 - include: ./scenarios/journal_collocation.yml
   when:
-    - (journal_collocation or dmcrypt_journal_collocation)
+    - (osd_scenario == 'journal_collocation' or osd_scenario == 'dmcrypt_journal_collocation')
```

Comment:
this is a good place for ``osd_scenario in ['journal_collocation', 'dmcrypt_journal_collocation']``